packages/e2e/test/blockfrost/queryTransactions.test.ts

```diff
@@ -0,0 +1,304 @@
+      env.CHAIN_HISTORY_PROVIDER_PARAMS,
+      logger
+    );
+  });
+
+  describe('transactionsByHashes', () => {
+    it('parses metadata correctly', async () => {
+      const [tx] = await chainHistoryProvider.transactionsByHashes({
+        ids: [Cardano.TransactionId('5127f8b235111854be744d36703f2098102d2639524035299658e73d683c0dfd')]
+      });
+      expect(tx.auxiliaryData!.blob!.get(1n)).toEqual(
+        new Map<string, string>([
+          ['absolute_slot', '67255007'],
+          ['timestamp', '1722938207']
+        ])
+      );
+    });
+
+    // TODO: waiting for resolution of
+    // https://github.com/blockfrost/openapi/issues/211
+    it.skip('parses collaterals correctly', async () => {
+      const [tx] = await chainHistoryProvider.transactionsByHashes({
+        ids: [Cardano.TransactionId('12fa9af65e21b36ec4dc4cbce478e911d52585adb46f2b4fe3d6563e7ee5a61a')]
+      });
+      expect(tx.body.inputs!.length).toEqual(1);
+      expect(tx.body.outputs!.length).toEqual(1);
+      expect(tx.body.collaterals!.length).toEqual(2);
+      expect(tx.body.collaterals![0]).toEqual({
+        address:
+          'addr_test1qrv0j69s06vd56365fsh5ync44ykaqr4exwf2vt6tuv76' +
+          'lcxy8jmk9rapjmuk2e7cfmshs27r4sx7tk0q3afktjf7j2qvdncx7',
+        index: 1,
+        txId: '2db6592c4782064295295b365c2e8ce84084fa24b1b3f5834f3c6b65268b7878'
+      });
+    });
+    // TODO: waiting for release/deploy in db-sync/blockfrost
+    // https://github.com/input-output-hk/cardano-db-sync/issues/1019
+    it.skip('has collaterals for failed contract', async () => {
+      const [tx] = await chainHistoryProvider.transactionsByHashes({
+        ids: [Cardano.TransactionId('9298f499a4c4aeba53a984cb4df0f9a93b7d158da4c2c2d12a06530841f94cd7')]
+      });
+      expect(tx.body.inputs!.length).toEqual(0);
+      expect(tx.body.outputs!.length).toEqual(0);
+      expect(tx.body.collaterals!.length).toEqual(1);
+    });
+    it('has withdrawals', async () => {
+      const [tx] = await chainHistoryProvider.transactionsByHashes({
+        ids: [Cardano.TransactionId('5929a59b9bebad1622f021d13b3d143d88cc92cf2400472e36ed8dcdf598a7fa')]
+      });
+      expect(tx.body.withdrawals).toEqual([
+        { quantity: 26_283_930n, stakeAddress: 'stake_test1urlhkh2pl2xt24dkgjtqrkzfv77ekqj950znqpzdsz2wuds0xlsk6' }
+      ]);
+    });
+    it('has redeemer', async () => {
+      const [tx] = await chainHistoryProvider.transactionsByHashes({
+        ids: [Cardano.TransactionId('f0c157c84506f5f3ead133133d6a731c937ebd0df379ed518fdbad9ee53ba28b')]
+      });
+
+      expect(tx.witness.redeemers).toBeDefined();
+      expect(tx.witness.redeemers).toHaveLength(1);
+      expect(tx.witness.redeemers![0].data).toBeDefined();
+      expect(tx.witness.redeemers![0]).toMatchObject({
+        executionUnits: { memory: 725_365, steps: 281_266_636 },
+        index: 0,
+        purpose: 'mint'
+      });
+    });
+    // TODO: not implemented
+    it.skip('has mint', async () => {
```

Comment:
Are we planning to merge these providers incomplete?

Reply:
@mkazlauskas my scope for this task was to revert the existing commits and make it work. 
I prefer to not merge.

Reply:
Let's convert this into a Draft PR then, so that it can't be merged